client/controller/controller.js

```diff
@@ -0,0 +1,355 @@
+
+app.controller('AppCtrl',function($scope, $http, filterdata) {
+
+
+    // ----> Code from autocomplete.js <------
+    var inputFrom = document.getElementById('input');
+
+
+    var autocompleteFrom = new google.maps.places.Autocomplete(inputFrom);
+    google.maps.event.addListener(autocompleteFrom, 'place_changed', function() {
+
+
+        var place = autocompleteFrom.getPlace();
+
+        var placeName = place.formatted_address;
+
+        $scope.place = placeName;
+
+
+
+    }); //end of autocomplete function.
+
+
+    //Funtion for the search button
+    // This function will recieve data from the server, filter the data , and respond back to the DOM
+    $scope.search = function() {
+
+     // newTyped();
+
+      //call the geocode function
+      $scope.geocode($scope.place);
+
+
+    };
+
+
+    $scope.geocode = function(placeName){
+
+
+      var geocoder = new google.maps.Geocoder();
+      geocoder.geocode({
+              address: placeName
+            }, function(results, status){
+              if(status === google.maps.GeocoderStatus.OK){
+
+                address = results[0];
+
+                  var location = {
+                    lat: address.geometry.location.lat(),
+                    lng: address.geometry.location.lng()
+                  };
+
+                var service = new google.maps.places.PlacesService(map.gMap);
+
+                var request = {
+                  location: location,
+                 // placeId: restaurant.place_id,
+                  type: ['restaurant'],
+                  radius: '10000'
+                };
+
+                service.nearbySearch(
+                 request,
+                 function(place, status) {
+                  if (status === google.maps.places.PlacesServiceStatus.OK) {
+
+                    //filterdata is an array of buisnesses.
+                    $scope.filterdata = filterdata.filter(place);
+
+
+
+                    //this will be used for the reset button
+                    $scope.resetR = $scope.filterdata;
+
+                    //call the getdetail method.
+                    $scope.getDetails();
+
+
+              }
+            })
+
+
+
+          }
+        });
+
+
+
+
+    } //end of geocode function
+
+
+    $scope.getDetails = function(){
+
+       document.getElementById('map-canvas').style.visibility="visible";
+      document.getElementById('slideshow').style.visibility = "visible";
+
+
+      var service = new google.maps.places.PlacesService(map.gMap);
+      var listOfRestaurants = $scope.filterdata;
+
+
+      var id = listOfRestaurants[0].place_id;
```

Comment:
Check if `listOfRestaurants` array > 0 else what is the scenario? error message?